x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_key_flyout.tsx

```diff
@@ -161,431 +162,560 @@ export const ApiKeyFlyout: FunctionComponent<ApiKeyFlyoutProps> = ({
-                    onChange={(value) => form.setValue('role_descriptors', value)}
-                    languageId="xjson"
-                    height={200}
-                    options={{ readOnly: readonly || (apiKey && !canEditApiKey) }}
-                  />
-                </EuiFormRow>
-                <EuiSpacer size="s" />
+                  <EuiFlexGroup gutterSize="m">
+                    <EuiFlexItem>
+                      <EuiCheckableCard
+                        id="rest"
+                        label={
+                          <>
+                            <EuiTitle size="xxs">
+                              <h2>
+                                <FormattedMessage
+                                  id="xpack.security.accountManagement.apiKeyFlyout.restTypeLabel"
+                                  defaultMessage="Personal API key"
+                                />
+                              </h2>
+                            </EuiTitle>
+                            <EuiSpacer size="xs" />
+                            <EuiText size="s">
+                              <FormattedMessage
+                                id="xpack.security.accountManagement.apiKeyFlyout.restTypeDescription"
+                                defaultMessage="Allow external services to access the Elastic Stack on your behalf."
+                              />
+                            </EuiText>
+                          </>
+                        }
+                        onChange={() => formik.setFieldValue('type', 'rest')}
+                        checked={formik.values.type === 'rest'}
+                      />
+                    </EuiFlexItem>
+                    <EuiFlexItem>
+                      <EuiCheckableCard
+                        id="cross_cluster"
+                        label={
+                          <>
+                            <EuiTitle size="xxs">
+                              <h2>
+                                <FormattedMessage
+                                  id="xpack.security.accountManagement.apiKeyFlyout.crossClusterTypeLabel"
+                                  defaultMessage="Cross-Cluster API key"
+                                />
+                              </h2>
+                            </EuiTitle>
+                            <EuiSpacer size="xs" />
+                            <EuiText size="s">
+                              <FormattedMessage
+                                id="xpack.security.accountManagement.apiKeyFlyout.crossClusterTypeDescription"
+                                defaultMessage="Allow remote clusters to connect to your local cluster."
+                              />
+                            </EuiText>
+                          </>
+                        }
+                        onChange={() => formik.setFieldValue('type', 'cross_cluster')}
+                        checked={formik.values.type === 'cross_cluster'}
```

Comment:
Minor nit: I'd consider making 'cross_cluster' and 'rest' constants. 'cross_cluster' is used a lot throughout the file for conditions. Just personal preference - I like working with constants because I don't always notice my own typos and it can lead to trouble.

Reply:
I've strongly typed these values so Typescript would flag up any typos. I think that gives us the protection we need while being less clunky to work with than constants.